### PR TITLE
[Feature] Support append columns in equality delete file

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

for iceberg v2, we need read the columns in equality delete file, so that we should append these columns in fe

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 40 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|25|00.00%|[890, 891, 892, 893, 894, 895, 896, 898, 899, 900, 901, 903, 904, 906, 907, 908, 909, 910, 911, 912, 913, 914, 916, 917, 918]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|0|15|00.00%|[63, 185, 186, 187, 188, 189, 190, 192, 193, 194, 195, 198, 199, 218, 222]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 31 (06.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|24|00.00%|[892, 893, 894, 895, 896, 897, 898, 900, 901, 902, 903, 905, 906, 908, 909, 910, 911, 912, 913, 914, 915, 916, 918, 919]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|2|7|28.57%|[193, 194, 195, 218, 222]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 31 (16.13%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|24|00.00%|[892, 893, 894, 895, 896, 897, 898, 900, 901, 902, 903, 905, 906, 908, 909, 910, 911, 912, 913, 914, 915, 916, 918, 919]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|5|7|71.43%|[217, 221]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
BE Build Success

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 27 / 31 (87.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|1|00.00%|[890]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|27|30|90.00%|[157, 158, 172]|

File: fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java [hunk 1]
```diff
@@ -122,6 +137,42 @@ private void preProcessConjuncts() {
...
+    public void appendEqualityColumns(PhysicalIcebergScanOperator node,
+                                      ColumnRefFactory columnRefFactory,
+                                      ExecPlan context) {
+        Table referenceTable = node.getTable();
+        Set<String> scanNodeColumns = node.getColRefToColumnMetaMap().values().stream()
+                .map(column -> column.getName()).collect(Collectors.toSet());
+        Set<String> appendEqualityColumns = equalityDeleteColumns.stream()
+                .filter(name -> !scanNodeColumns.contains(name)).collect(Collectors.toSet());
+        for (String eqName : appendEqualityColumns) {
+            for (Column column : referenceTable.getFullSchema()) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java:152]: why not use hashmap, if there are 1000 column, the cost is high

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 27 / 30 (90.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|1|00.00%|[890]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|27|29|93.10%|[159, 160]|

**Review by Youngwb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java [hunk 2]
```diff
@@ -184,6 +235,12 @@ public void getScanRangeLocations() throws UserException {
                             source.content() == FileContent.EQUALITY_DELETES ? TIcebergFileContent.EQUALITY_DELETES : TIcebergFileContent.POSITION_DELETES);
                     target.setLength(source.fileSizeInBytes());
 
+                    if (source.content() == FileContent.EQUALITY_DELETES) {
+                        source.equalityFieldIds().stream().forEach(fieldId -> {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java:239]: nit : source.equalityFieldIds().forEach ...

File: fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java [hunk 3]
```diff
@@ -122,6 +138,41 @@ private void preProcessConjuncts() {
...
+     */
+    public void appendEqualityColumns(PhysicalIcebergScanOperator node,
+                                      ColumnRefFactory columnRefFactory,
+                                      ExecPlan context) {
+        Table referenceTable = node.getTable();
+        Set<String> scanNodeColumns = node.getColRefToColumnMetaMap().values().stream()
+                .map(column -> column.getName()).collect(Collectors.toSet());
+        Set<String> appendEqualityColumns = equalityDeleteColumns.stream()
+                .filter(name -> !scanNodeColumns.contains(name)).collect(Collectors.toSet());
+        Map<String, Object> nameToColumns = referenceTable.getFullSchema().stream().collect(Collectors.toMap(Column::getName, item -> item));
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java:152]: nit:
1. There are over 130 in this line. 
2. Map<String, Object> -> Map<String, Column>

File: fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java [hunk 4]
```diff
@@ -122,6 +138,41 @@ private void preProcessConjuncts() {
...
+        for (String eqName : appendEqualityColumns) {
+            if (nameToColumns.containsKey(eqName)) {
+                Column column = (Column) nameToColumns.get(eqName);
+                Field field;
+                TableName tableName = desc.getRef().getName();
+                if (referenceTable.getBaseSchema().contains(column)) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java:158]: It's better to use the same function "getBaseSchema" or "getFullSchema" with line 152.

**Review by stephen-shelby — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Review by dirtysalt — DISMISSED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by caneGuy** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 28 / 31 (90.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|1|00.00%|[890]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|28|30|93.33%|[163, 164]|

**Comment by caneGuy** [comment]
run starrocks_admit_test

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**